File: server/src/Routes/ActivityRoutes.js
import express from "express";
import JWT from "../Middlewares/jsonwebtoken.js";
import ActivityController from "../Controllers/activityController.js";

const ActivityRoutes = express.Router();

ActivityRoutes.get(
  "/get-all",
  JWT.validateAccessToken,
  ActivityController.getAllActivities
);

ActivityRoutes.get(
  "/get-one/:activityId",
  JWT.validateAccessToken,
  ActivityController.getActivity
);
ActivityRoutes.post(
  "/create",
  JWT.validateAccessToken,
  ActivityController.createActivity
);
ActivityRoutes.put(
  "/update/:activityId",
  JWT.validateAccessToken,
  ActivityController.updateActivity
);
ActivityRoutes.delete(
  "/delete/:activityId",
  JWT.validateAccessToken,
  ActivityController.deleteActivity
);

export default ActivityRoutes;

File: server/src/routes/activityRoutes.go
package routes

import (
	"controll/src/controllers"

	"github.com/gin-gonic/gin"
)

func ActivityRoutes(router *gin.RouterGroup) {
	activity := router.Group("/api/activity")
	{
		activity.GET("/get-all", controllers.GetAllActivities())
		activity.GET("/get-one/:activityId", controllers.GetActivity())
		activity.POST("/create", controllers.CreateActivity())
		activity.PUT("/update/:activityId", controllers.UpdateActivity())
		activity.DELETE("/delete/:activityId", controllers.DeleteActivity())
	}
}
